src/main/java/io/confluent/connect/jdbc/dialect/GenericDatabaseDialect.java

```diff
@@ -0,0 +1,1727 @@
+        if (optional) {
+          tsSchemaBuilder.optional();
+        }
+        builder.field(fieldName, tsSchemaBuilder.build());
+        break;
+      }
+
+      case Types.ARRAY:
+      case Types.JAVA_OBJECT:
+      case Types.OTHER:
+      case Types.DISTINCT:
+      case Types.STRUCT:
+      case Types.REF:
+      case Types.ROWID:
+      default: {
+        log.warn("JDBC type {} ({}) not currently supported", sqlType, columnDefn.typeName());
+        return null;
+      }
+    }
+    return fieldName;
+  }
+
+  @Override
+  public void applyDdlStatements(
+      Connection connection,
+      List<String> statements
+  ) throws SQLException {
+    try (Statement statement = connection.createStatement()) {
+      for (String ddlStatement : statements) {
+        statement.executeUpdate(ddlStatement);
+      }
+    }
+  }
+
+  @Override
+  public ColumnConverter createColumnConverter(
+      ColumnMapping mapping
+  ) {
+    return columnConverterFor(mapping, mapping.columnDefn(), mapping.columnNumber(),
+                              jdbcDriverInfo().jdbcVersionAtLeast(4, 0)
+    );
+  }
+
+  @SuppressWarnings("deprecation")
+  protected ColumnConverter columnConverterFor(
+      final ColumnMapping mapping,
+      final ColumnDefinition defn,
+      final int col,
+      final boolean isJdbc4
+  ) {
+    switch (mapping.columnDefn().type()) {
+
+      case Types.BOOLEAN: {
+        return (rs) -> rs.getBoolean(col);
+      }
+
+      case Types.BIT: {
+        /**
+         * BIT should be either 0 or 1.
+         * TODO: Postgres handles this differently, returning a string "t" or "f". See the
+         * elasticsearch-jdbc plugin for an example of how this is handled
+         */
+        return (rs) -> rs.getByte(col);
+      }
+
+      // 8 bits int
+      case Types.TINYINT: {
+        if (defn.isSignedNumber()) {
+          return (rs) -> rs.getByte(col);
+        } else {
+          return (rs) -> rs.getShort(col);
+        }
+      }
+
+      // 16 bits int
+      case Types.SMALLINT: {
+        if (defn.isSignedNumber()) {
+          return (rs) -> rs.getShort(col);
+        } else {
+          return (rs) -> rs.getInt(col);
+        }
+      }
+
+      // 32 bits int
+      case Types.INTEGER: {
+        if (defn.isSignedNumber()) {
+          return (rs) -> rs.getInt(col);
+        } else {
+          return (rs) -> rs.getLong(col);
+        }
+      }
+
+      // 64 bits int
+      case Types.BIGINT: {
+        return (rs) -> rs.getLong(col);
+      }
+
+      // REAL is a single precision floating point value, i.e. a Java float
+      case Types.REAL: {
+        return (rs) -> rs.getFloat(col);
+      }
+
+      // FLOAT is, confusingly, double precision and effectively the same as DOUBLE. See REAL
+      // for single precision
+      case Types.FLOAT:
+      case Types.DOUBLE: {
+        return (rs) -> rs.getDouble(col);
+      }
+
+      case Types.NUMERIC:
```

Comment:
@kkonstantine this is the second of two blocks which incorporate the substantiative portions of your recent changes from #396. Would love a sanity check here.